jmespath-core/src/main/java/io/burt/jmespath/parser/ExpressionParser.java

```diff
@@ -85,7 +85,7 @@ private String identifierToString(JmesPathParser.IdentifierContext ctx) {
   private void checkForUnescapedBackticks(Token token) {
     int unescapedBacktickIndex = indexOfUnescapedBacktick(token.getText());
     if (unescapedBacktickIndex > -1) {
-      errors.parseError("unexpected `", token.getLine(), token.getStartIndex() + unescapedBacktickIndex);
+      errors.parseError("syntax error unexpected `", token.getLine(), token.getStartIndex() + unescapedBacktickIndex);
```

Comment:
In principle, `syntax error unexpected` reads a bit awkwardly, which I guess you could solve by introducing a bridging word(s) such as `at`, `due to`, `caused by` or something similar before unexpected.